src/Surfnet/StepupMiddleware/ApiBundle/Identity/Entity/RaSecondFactor.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddleware\ApiBundle\Identity\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use JsonSerializable;
+
+/**
+ * A second factor as displayed in the registration authority application. One exists for every second factor,
+ * regardless of state. As such, it sports a status property, indicating whether its vetted, revoked etc.
+ *
+ * @ORM\Entity(repositoryClass="Surfnet\StepupMiddleware\ApiBundle\Identity\Repository\RaSecondFactorRepository")
+ * @ORM\Table(
+ *      indexes={
+ *          @ORM\Index(name="idx_ra_second_factor_type", columns={"type"}),
+ *          @ORM\Index(name="idx_ra_second_factor_second_factor_id", columns={"second_factor_id"}),
+ *          @ORM\Index(name="idx_ra_second_factor_status", columns={"status"}),
+ *          @ORM\Index(name="idx_ra_second_factor_identity_id", columns={"identity_id"}),
+ *          @ORM\Index(name="idx_ra_second_factor_institution", columns={"institution"}),
+ *          @ORM\Index(name="idx_ra_second_factor_name", columns={"name"}),
+ *          @ORM\Index(name="idx_ra_second_factor_email", columns={"email"}),
```

Comment:
Not quite sure how I feel about the indexes... Esp. `status` and `type` might not benefit from an index due to lack of different values, but does incur the cost of having to maintain the index. Furthermore it seems that `name` and `email` would benefit from a `FULLTEXT` index. Also iirc not all these fields are searchable and thus would not need an index? Lastly, compound indexes covering the most common usecases would be very beneficial. Thoughts?